helix-view/src/input.rs

```diff
@@ -4,8 +4,54 @@ use helix_core::unicode::{segmentation::UnicodeSegmentation, width::UnicodeWidth
 use serde::de::{self, Deserialize, Deserializer};
 use std::fmt;
 
-use crate::keyboard::{KeyCode, KeyModifiers};
+pub use crate::keyboard::{KeyCode, KeyModifiers};
+
+#[derive(Debug, PartialOrd, PartialEq, Eq, Clone, Copy, Hash)]
+pub enum Event {
+    Key(KeyEvent),
+    Mouse(MouseEvent),
+    Resize(u16, u16),
+    IdleTimeout,
+}
```

Comment:
I think this would make more sense as a nested type:

```rust
enum Event {
   Input(InputEvent),
   IdleTimeout,
}
```

Reply:
I didn't nest the type to avoid noise in the change set, and seems like `FocusGained` and `FocusLost` have also been introduced as new Event types. Should we still nest, and if so, should the focus events be moved out of `Event::Input` and into a top level `Event::Focus(FocusEvent)` or simply leave them in `InputEvent` and only factor out `IdleTimeout`?

Reply:
Hmm okay, let's leave it as is for now